### PR TITLE
override slimPatterns instead of appending

### LINKED ISSUE #216: Allow ability to override slim defaults
(Note: I am packaging individually with `zip: true`, and am using a branch from @felipe-augusto to deal with the issue as described in #203.)

In order to meet lambda size limitations I am using `slim: true`, however I was not able to successfully run due to one my modules (Sagemaker SDK) not being able to find a dependency (in this case urllib3).  What I found is that Sagemaker is using pkg_resources to pull the SDK version (below), and then pkg_resources throws an exception because it is not able to identify the distribution information since all of the dist-info files have been removed by slim.

`SDK_VERSION = pkg_resources.require('sagemaker')[0].version`

Setting `slim: false` is not an option for me since I need that additional trimming, but there is no easy workaround since custom slim options only append to the defaults.  I could create specialized packages that I trim myself, or create a branch that removes that default option.  I went with the latter, but I think it would be a handy feature to allow for overriding the defaults.  If defining custom slim options, it might be easier to remove the defaults altogether, and if a user wants them they can add them back in manually or use a special `default` option.

### PR DESCRIPTION
**Pull request by bweigel**
closes #216

### CODE REVIEW

**Comment by bweigel** [comment]
@dschep I added a new option `slimPatternsAppendDefaults`, which defaults to `true`. This keeps backwards compatibility and when set to `false` adds the possibility to overwrite the default patterns with just the custom patterns.

**Comment by bweigel** [comment]
I'll look into the tests later. I am not very knowledgable when it comes to node & testing, so I'll have to look into it.

**Comment by dschep** [comment]
don't worry if you can't figure it out. just let me know and i'll take care of it.

**Comment by dschep** [comment]
one more thing, could you run `yarn format` and commit the changes & push. Thanks :)

**Comment by bweigel** [comment]
I have migrated the tests from bats to `test.js`. However, I have not implemented checks for Docker and I guess some of these tests fail, because docker is missing in the AppVeyor CI system.

**Comment by bweigel** [comment]
@dschep do you have some clue or hints on how to deal with the failing tests (i.e. docker is missing in the AppVeyor CI system)?

I am guessing in bats these tests are always skipped via `docker &> /dev/null || skip "docker not present"`. I don't know how to deal with that in the `test.js`-tests.

**Comment by dschep** [comment]
Here's a function to check if Docker can be used(this should actually cover all the use cases that bats was checking for) in JS:
```javascript
const canUseDocker = () => {
  let result
  try {
    result = spawnSync('docker', ['ps']);
  } catch (e) {
    return false;
  }
  return result.status === 0;
}

```

File: test.js [hunk 1]
```diff
@@ -164,7 +171,141 @@ test("py3.6 doesn't package bottle with zip option", t => {
...
+  );
+  t.deepEqual(
+    zipfiles.filter(filename => filename.includes('.egg-info')),
+    [],
+    '.egg-info folders are not packaged'
+  );
+  t.end();
+});
+
+test('py3.6 can package flask with slim & dockerizePip & slimPatterns & slimPatternsAppendDefaults=false options', t => {
```
Comment (dschep) [inline test.js]:
```suggestion
test('py3.6 can package flask with slim & dockerizePip & slimPatterns & slimPatternsAppendDefaults=false options', {skip: !canUseDocker()}, t => {
```

File: test.js [hunk 2]
```diff
@@ -86,7 +86,7 @@ const listRequirementsZipFiles = filename => {
   const zip = deasync(new JSZip().loadAsync(readFileSync(filename)));
   const reqsBuffer = deasync(zip.file('.requirements.zip').async('nodebuffer'));
   const reqsZip = deasync(new JSZip().loadAsync(reqsBuffer));
-  return Object.keys(reqsZip.files)
+  return Object.keys(reqsZip.files);
 };
```
Comment (dschep) [inline test.js]:
```suggestion

const canUseDocker = () => {
  let result
  try {
    result = crossSpawn.sync('docker', ['ps']);
  } catch (e) {
    return false;
  }
  return result.status === 0;
}

```

**Comment by bweigel** [comment]
sweet. thanks!

**Pull request merged**